gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -87,11 +87,14 @@ hook.Add("Think", "TTT2PlayerSprinting", function()
 		if not ply.sprintTS then continue end
 
 		local timeElapsed = CurTime() - ply.sprintTS
+		local isSprinting = (ply.sprintMultiplier ~= nil)
 
-		if not ply.sprintMultiplier then
-			ply.sprintProgress = math.min((ply.oldSprintProgress or 0) + timeElapsed * GetGlobalFloat("ttt2_sprint_stamina_regeneration"), 1)
+		local drainMult, regenMult = hook.Run("TTT2StaminaMult", ply, isSprinting)
```

Comment:
@saibotk I really don't like this solution, because this seems to be inconsistent. Every time, one var is not used and all the time nil. Using this hook seems to be tricky too. I think one var should be enough. The only thing that makes sense would be to call a "TTT2StaminaDrainMult" and a "TTT2StaminaRegenMult" hook some lines below. Then we can remove the isSprinting value and everything should be more consistent with better integrity.

Reply:
The separation of vars was suggested by @saibotk earlier in Discord, and I did initially plan to only keep one variable. Would calling it staminaMult be a good fit?

Reply:
No, thats not what @Alf21 means, they should be separate but he dislikes the combined hook, which can be split up in 2 hooks, i just suggested it with the thought of not adding too many useless hooks, but i suppose this should be okay here.